test/common/common/utility_test.cc

```diff
@@ -35,6 +35,33 @@ TEST(StringUtil, atoul) {
   EXPECT_EQ(789U, out);
 }
 
+TEST(StringUtil, atol) {
+  int64_t out;
+  EXPECT_FALSE(StringUtil::atol("-123b", out));
+  EXPECT_FALSE(StringUtil::atol("", out));
+  EXPECT_FALSE(StringUtil::atol("b123", out));
+
+  EXPECT_TRUE(StringUtil::atol("123", out));
+  EXPECT_EQ(123, out);
+  EXPECT_TRUE(StringUtil::atol("-123", out));
+  EXPECT_EQ(-123, out);
+  EXPECT_TRUE(StringUtil::atol("+123", out));
+  EXPECT_EQ(123, out);
+
+  EXPECT_TRUE(StringUtil::atol("  456", out));
+  EXPECT_EQ(456, out);
+
+  EXPECT_TRUE(StringUtil::atol("00789", out));
+  EXPECT_EQ(789, out);
+
+  // INT64_MAX + 1
+  EXPECT_FALSE(StringUtil::atol("9223372036854775808", out));
+
+  // INT64_MIN
+  EXPECT_TRUE(StringUtil::atol("-9223372036854775808", out));
+  EXPECT_EQ(-9223372036854775808, out);
```

Comment:
Apparently this is being treated as the negation of 9223372036854775808. I think using INT64_MIN here would be fine.